td-shim-tools/src/tee_info_hash.rs

```diff
@@ -243,11 +242,9 @@ impl TdInfoStruct {
             .seek(SeekFrom::Start(metadata_off as u64))
             .unwrap();
 
-        let mut metadata_buf = [0; TD_SHIM_METADATA_SIZE as usize];
-
-        raw_image_file.read_exact(&mut metadata_buf).unwrap();
-
-        let desc = &metadata_buf[size_of::<TdxMetadataGuid>()..];
+        let mut desc_buf = [0; size_of::<TdxMetadataGuid>() + size_of::<TdxMetadataDescriptor>()];
+        raw_image_file.read_exact(&mut desc_buf).unwrap();
+        let desc = &desc_buf[size_of::<TdxMetadataGuid>()..];
```

Comment:
I am a little confused. The logic here means
1. The memory starts from `metadata_off` is a `TdxMetadataGuid` and a `TdxMetadataDescriptor`
2. read the `TdxMetadataGuid` and `TdxMetadataDescriptor`
3. `desc` is assigned with the memory slice of `TdxMetadataDescriptor`

Reply:
The previous code assumed the buffer size is TD_SHIM_METADATA_SIZE which is 0x1000. So it read from metadata_off to metadata_off+0x1000. 
But actually it should read TdxMetadataDescriptor first and get the length from TdxMetadataDescriptor structure and set is as buffer size.